player/lua/ytdl_hook.lua

```diff
@@ -731,6 +759,11 @@ function run_ytdl_hook(url)
         end
         if (param == "sub-lang") and (arg ~= "") then
             allsubs = false
+        elseif (param == "write-thumbnail") then
```

Comment:
I know this is old and no idea if you're still interested in working on this, but I think this is the wrong approach. It shouldn't be tied to any yt-dlp/youtube-dl option. Fetching thumbnails should be a new ytdl_hook option. This is a cool feature though.